lib/ansible/modules/windows/win_nssm.py

```diff
@@ -58,7 +58,7 @@
       - Use either this or C(app_parameters), not both.
   dependencies:
     description:
-      - Service dependencies that has to be started to trigger startup, separated by comma.
+      - Service dependencies that has to be started to trigger startup, separated by space.
```

Comment:
I'm not sure if separating by a space is better than a comma, a service name can definitely have a space and this break backwards compatibility.

Instead we should change the type of dependencies to be a list which allows people to define it like

```
dependencies
- adf
- tcpip
```
This means they user doesn't have to deal with escaping issues for complex service names as well as keeping backwards compatibilities as the code already splits by `,`. This is all done in `Get-AnsibleParam` so no logic is required in win_nssm.

Reply:
Neither space nor comma are good separators in fact, as a service name can also have a comma (and probably any printable ASCII character except slash and backslashes):
![image](https://user-images.githubusercontent.com/3837104/44777244-6560c280-ab7a-11e8-83ce-56c0ea71d6c9.png)

But again here is a simple and backportable fix to get current and previous versions to work almost well, because the current code just doesn't work if you use a comma as separator (the module pass the string as is to NSSM and an error is thrown saying that the service "service1,service2" does not exist). So from my point of view this doesn't break BC.

I also thought about the list form and I think it will be much better for the future, but this may be part of following PRs. I actually prepared the code in this PR for this purpose, as I already split the string into an array to be able to do a order insensitive comparison:
https://github.com/ansible/ansible/blob/51843a7b3c7120721d19b192408d7a5984b12d67/lib/ansible/modules/windows/win_nssm.ps1#L434-L437

Reply:
Yep agreed that services can have spaces or commas, this is why setting the type for this in the `Get-AnsibleParam` to `list` means someone can define a list in the playbook side and not have to worry about that. While it "may" not work today as a comma separated string, Legacy.psm1 has internal logic to split a string by `,` if `type=list`. Having it split by a space goes against this logic and really doesn't need to be done in win_nssm.

Reply:
Ok, after a second look, I understand your point, and you're perfetly right. Do we need to change something in the doc to specify that a list is accepted ?

By the way, it makes me find another issue with this module: if you give a name containing a quote to your service, the module fails because the quote is not escaped in the commands run by the module...

```yml
- win_nssm:
    name: test"quote
    application: 'C:\Windows\System32\cmd.exe'
```

In a future PR, I think we need to consider refactoring the code more in depth so that all the command-line parameters go through the Escape-Argument/Argv-ToString functions, as other module parameters may be affected too.